### PR TITLE
[WIP] ControlParameter inordinatly slow on big file.

### PR DESCRIPTION
**Pull request by troessner**
WIP for #1381

### CODE REVIEW

File: spec/performance/reek/smell_detectors/control_parameter_spec.rb [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+require_lib 'reek/examiner'
+require 'stackprof'
+
+RSpec.describe Reek::SmellDetectors::ControlParameter do
+  let(:source) { SAMPLES_DIR.join('smelly_source').join('ruby.rb') }
+
+  it 'runs on a complicated source file in less than 10 seconds' do
+    examiner = Reek::Examiner.new source, filter_by_smells: ['ControlParameter']
+    # This is the actual test
+    # expect(examiner.smells.size).to perform_under(10).sec
```
Comment (troessner) [inline spec/performance/reek/smell_detectors/control_parameter_spec.rb]: @mvz stumbled across [this gem](https://github.com/piotrmurach/rspec-benchmark) - seems worth a shot to me. Also wdyt about those kind of performance specs in general?

File: Gemfile [hunk 2]
```diff
@@ -12,14 +12,13 @@ group :development do
...
   gem 'rake',          '~> 12.0'
   gem 'rspec',         '~> 3.0'
+  gem 'rspec-benchmark'
+  gem 'rubocop',       '~> 0.57.1'
+  gem 'rubocop-rspec', '~> 1.20'
   gem 'simplecov',     '~> 0.16.1'
+  gem 'stackprof',     '~> 0.2.1'
   gem 'yard',          '~> 0.9.5'
 
-  if RUBY_VERSION >= '2.3'
```
Comment (mvz) [inline Gemfile]: :boom: :tada:

File: spec/performance/reek/smell_detectors/control_parameter_spec.rb (hunk 1 above)
Comment (mvz) [reply spec/performance/reek/smell_detectors/control_parameter_spec.rb]: Yes, definitely worth a try. I have no experience with this kind of spec but it looks very useful.

**Comment by troessner** [comment]
Problem solved via #1389, closing this one.

**Pull request closed without merging**